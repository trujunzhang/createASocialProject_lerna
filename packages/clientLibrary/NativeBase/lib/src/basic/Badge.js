import React, { Component } from "react";
import PropTypes from "prop-types";
import { View, ViewPropTypes } from "react-native";

import { connectStyle } from "@app/native-base-shoutem-theme";
import { platformVariables as variable } from '@app/native-base-variables';
import mapPropsToStyleNames from "../utils/mapPropsToStyleNames";

class Badge extends Component {
  render() {
    return (
      <View ref={c => (this._root = c)} {...this.props}>
        {this.props.children}
      </View>
    );
  }
}

Badge.propTypes = {
  ...ViewPropTypes,
  style: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.number,
    PropTypes.array
  ])
};

const StyledBadge = connectStyle("NativeBase.Badge", {}, mapPropsToStyleNames)(
  Badge
);

export { StyledBadge as Badge };
