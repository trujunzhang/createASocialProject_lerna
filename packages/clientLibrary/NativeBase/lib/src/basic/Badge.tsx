import * as React from 'react'
import PropTypes from "prop-types";
import { View, ViewPropTypes } from "react-primitives";

import { connectStyle } from "@app/native-base-shoutem-theme";
import { platformVariables as variable } from '@app/native-base-variables';
import mapPropsToStyleNames from "../utils/mapPropsToStyleNames";

class Badge extends React.Component {
  private _root: any
  render() {
    return (
      <View ref={c => (this._root = c)} {...this.props}>
        {this.props.children}
      </View>
    );
  }
}

// Badge.propTypes = {
//   ...ViewPropTypes,
//   style: PropTypes.oneOfType([
//     PropTypes.object,
//     PropTypes.number,
//     PropTypes.array
//   ])
// };

const StyledBadge = connectStyle("NativeBase.Badge", {}, mapPropsToStyleNames)(
  Badge
);

export { StyledBadge as Badge };
