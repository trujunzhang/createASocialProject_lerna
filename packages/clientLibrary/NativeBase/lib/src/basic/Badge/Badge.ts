import * as React from 'react'

import { ReactNative, BsStyle } from '@appUtils/rnCommonProps'

export interface BadgeProps extends ReactNative.ViewProps, BsStyle {
  // Others
  textStyle?: any
}
