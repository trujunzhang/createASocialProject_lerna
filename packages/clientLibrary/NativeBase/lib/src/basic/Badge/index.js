import * as React from 'react';
import { View } from 'react-primitives';
import { connectStyle } from 'native-base-shoutem-theme';
import { mapPropsToStyleNames } from '@app/tools';
class Badge extends React.Component {
    render() {
        return (React.createElement(View, Object.assign({ ref: (c) => (this._root = c) }, this.props), this.props.children));
    }
}
// Badge.propTypes = {
//   ...ViewPropTypes,
//   style: PropTypes.oneOfType([
//     PropTypes.object,
//     PropTypes.number,
//     PropTypes.array
//   ])
// };
const StyledBadge = connectStyle('NativeBase.Badge', {}, mapPropsToStyleNames)(Badge);
export { StyledBadge as Badge };
//# sourceMappingURL=index.js.map