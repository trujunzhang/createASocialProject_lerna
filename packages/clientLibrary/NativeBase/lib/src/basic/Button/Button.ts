import * as React from 'react'

import { ReactNative, RnViewStyleProp, RnTextStyleProp, BsStyle } from '@appUtils/rnCommonProps'

/**
 * see Widget Button.js
 */
export interface ButtonProps extends ReactNative.TouchableOpacityProps, BsStyle {
  /**
   * Defines button style
   */
  style?: RnViewStyleProp | Array<RnViewStyleProp>
  /**
   * Defines button text style
   */
  textStyle?: RnTextStyleProp
  // onPress?: () => void
  /**
   * Block level button
   */
  block?: boolean
  /**
   * Vertical button
   */
  vertical?: boolean
  /**
   * Button with badges
   */
  badge?: boolean
  //primary?: boolean,
  /**
   * Gives you effect of Icon-buttons.
   * To have button with transparent background, include this prop.
   */
  transparent?: boolean
  //success?: boolean,
  //danger?: boolean,
  // warning?: boolean,
  //info?: boolean,
  color?: string
  /**
   * Applies outline button style.
   */
  bordered?: boolean
  /**
   * Renders button with slightly round shaped edges.
   */
  rounded?: boolean
  /**
   * For large size button
   */
  large?: boolean
  /**
   * For small size button
   */
  small?: boolean
  /**
   * Used for Icon alignment.
   * Aligns icon to the left in button.
   * By default, icons are aligned to the left in button.
   */
  iconLeft?: boolean
  /**
   * Used for Icon alignment.
   * Aligns icon to the right in button.
   */
  iconRight?: boolean
  /**
   * Used to enable/disable icons
   * Icons align in the center of the button.
   */
  icon?: boolean
  /**
   * Disables onPress option for button
   */
  disabled?: boolean
  active?: boolean
  inputButton?: boolean
  full?: boolean
  light?: boolean
  dark?: boolean
  /**
   * [android] colored ripple effect
   */
  androidRippleColor?: string
  /**
   * Segments
   */
  first?: boolean
  last?: boolean
  /**
   * Adds necessary padding when Text button defined in Left / Right of Header (iOS)
   */
  hasText?: boolean
  // Others
  activeOpacity?: any
  onlyCenter?: boolean
  facebook?: boolean
  twitter?: boolean
  svgLeft?: boolean
}
