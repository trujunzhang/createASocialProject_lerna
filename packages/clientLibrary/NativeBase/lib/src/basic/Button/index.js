import * as React from 'react';
import * as PropTypes from 'prop-types';
import { TouchableOpacity, Platform, View, TouchableNativeFeedback, StyleSheet } from 'react-primitives';
import { connectStyle } from 'native-base-shoutem-theme';
import { platformVariables as variable } from '@appVariables/theme/variables/platform'; // variables // [variable]
import { Text } from '@appBasic/Text';
import computeProps from '@appUtils/computeProps';
import { mapPropsToStyleNames } from '@app/tools';
class Button extends React.Component {
    getInitialStyle() {
        return {
            borderedBtn: {
                borderWidth: this.props.bordered ? 1 : undefined,
                borderRadius: this.props.rounded && this.props.bordered ? variable.borderRadiusLarge : 2
            }
        };
    }
    prepareRootProps() {
        const defaultProps = {
            style: this.getInitialStyle().borderedBtn
        };
        if (Array.isArray(this.props.style)) {
            const flattenedStyle = this.props.style.reduce((accumulator, currentValue) => accumulator.concat(currentValue), []);
            return computeProps({ ...this.props, style: flattenedStyle }, defaultProps);
        }
        return computeProps(this.props, defaultProps);
    }
    render() {
        const variables = this.context.theme
            ? this.context.theme['@@shoutem.theme/themeStyle'].variables
            : variable;
        const children = Platform.OS === 'ios'
            ? this.props.children
            : React.Children.map(this.props.children, (child) => child && child.type === Text
                ? React.cloneElement(child, {
                    uppercase: variables.btnUppercaseAndroidText,
                    ...child.props
                })
                : child);
        if (Platform.OS === 'ios' ||
            Platform.OS === 'web' ||
            variables.androidRipple === false ||
            Platform['Version'] < 21) {
            return (React.createElement(TouchableOpacity, Object.assign({}, this.prepareRootProps(), { ref: (c) => (this._root = c), activeOpacity: this.props.activeOpacity > 0 ? this.props.activeOpacity : 0.5 }), children));
        }
        else {
            if (this.props.rounded) {
                let buttonStyle = { ...this.prepareRootProps().style };
                let buttonFlex = this.props.full || this.props.block ? 1 : buttonStyle.flex;
                return (React.createElement(View, { style: [
                        { maxHeight: buttonStyle.height },
                        buttonStyle,
                        { paddingTop: undefined, paddingBottom: undefined }
                    ] },
                    React.createElement(TouchableNativeFeedback, Object.assign({ ref: (c) => (this._root = c), background: this.props.androidRippleColor
                            ? TouchableNativeFeedback.Ripple(this.props.androidRippleColor, true)
                            : TouchableNativeFeedback.Ripple(variables.androidRippleColor, true) }, this.prepareRootProps()),
                        React.createElement(View, { style: [
                                styles.childContainer,
                                {
                                    paddingTop: buttonStyle.paddingTop,
                                    paddingBottom: buttonStyle.paddingBottom,
                                    height: buttonStyle.height,
                                    flexGrow: buttonFlex
                                }
                            ] }, children))));
            }
            else {
                return (React.createElement(TouchableNativeFeedback, Object.assign({ ref: (c) => (this._root = c), onPress: this.props.onPress, background: this.props.transparent
                        ? TouchableNativeFeedback.Ripple('transparent')
                        : TouchableNativeFeedback.Ripple(variables.androidRippleColor, false) }, this.prepareRootProps()),
                    React.createElement(View, Object.assign({}, this.prepareRootProps()), children)));
            }
        }
    }
}
Button.contextTypes = {
    theme: PropTypes.object
};
// Button.propTypes = {
//   ...TouchableOpacity.propTypes,
//   style: PropTypes.oneOfType([
//     PropTypes.object,
//     PropTypes.number,
//     PropTypes.array
//   ]),
//   block: PropTypes.bool,
//   primary: PropTypes.bool,
//   transparent: PropTypes.bool,
//   success: PropTypes.bool,
//   danger: PropTypes.bool,
//   warning: PropTypes.bool,
//   info: PropTypes.bool,
//   bordered: PropTypes.bool,
//   disabled: PropTypes.bool,
//   rounded: PropTypes.bool,
//   large: PropTypes.bool,
//   small: PropTypes.bool,
//   active: PropTypes.bool
// };
const styles = StyleSheet.create({
    childContainer: {
        flexShrink: 1,
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center'
    }
});
const StyledButton = connectStyle('NativeBase.Button', {}, mapPropsToStyleNames)(Button);
export { StyledButton as Button };
//# sourceMappingURL=index.js.map