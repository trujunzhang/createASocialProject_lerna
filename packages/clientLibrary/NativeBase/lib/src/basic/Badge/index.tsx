import * as React from 'react'
import * as PropTypes from 'prop-types'
import { View } from 'react-primitives'

import { connectStyle } from 'native-base-shoutem-theme'

import { IThemeVariables } from '@app/tools' // typings

import { platformVariables as variables } from '@appVariables/theme/variables/platform' // variables // [variables]
import { mapPropsToStyleNames } from '@app/tools'

import { BadgeProps } from './Badge'

class Badge extends React.Component<BadgeProps, any> {
  private _root: any
  render() {
    return (
      <View ref={(c) => (this._root = c)} {...this.props}>
        {this.props.children}
      </View>
    )
  }
}

// Badge.propTypes = {
//   ...ViewPropTypes,
//   style: PropTypes.oneOfType([
//     PropTypes.object,
//     PropTypes.number,
//     PropTypes.array
//   ])
// };

const StyledBadge = connectStyle('NativeBase.Badge', {}, mapPropsToStyleNames)(Badge)

export { StyledBadge as Badge }
