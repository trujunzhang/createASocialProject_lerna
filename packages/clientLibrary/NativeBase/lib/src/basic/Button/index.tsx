import * as React from 'react'
import * as PropTypes from 'prop-types'
import {
  TouchableOpacity,
  Platform,
  View,
  TouchableNativeFeedback,
  StyleSheet
} from 'react-primitives'
import { connectStyle } from 'native-base-shoutem-theme'
import { IThemeVariables } from '@app/tools' // typings
import { platformVariables as variable } from '@appVariables/theme/variables/platform' // variables // [variable]

import { Text } from '@appBasic/Text'
import computeProps from '@appUtils/computeProps'

import { mapPropsToStyleNames } from '@app/tools'

import { ButtonProps } from './Button'

class Button extends React.Component<ButtonProps, any> {
  static contextTypes = {
    theme: PropTypes.object
  }

  getInitialStyle() {
    return {
      borderedBtn: {
        borderWidth: this.props.bordered ? 1 : undefined,
        borderRadius: this.props.rounded && this.props.bordered ? variable.borderRadiusLarge : 2
      }
    }
  }

  // _root: React$Element<TouchableOpacity | TouchableNativeFeedback>;
  _root: any

  prepareRootProps() {
    const defaultProps = {
      style: this.getInitialStyle().borderedBtn
    }

    if (Array.isArray(this.props.style)) {
      const flattenedStyle = this.props.style.reduce(
        (accumulator: any, currentValue) => accumulator.concat(currentValue),
        []
      )
      return computeProps({ ...this.props, style: flattenedStyle }, defaultProps)
    }

    return computeProps(this.props, defaultProps)
  }
  render() {
    const variables: IThemeVariables = this.context.theme
      ? this.context.theme['@@shoutem.theme/themeStyle'].variables
      : variable
    const children =
      Platform.OS === 'ios'
        ? this.props.children
        : React.Children.map(this.props.children, (child: any) =>
            child && child.type === Text
              ? React.cloneElement(child, {
                  uppercase: variables.btnUppercaseAndroidText,
                  ...child.props
                })
              : child
          )
    if (
      Platform.OS === 'ios' ||
      Platform.OS === 'web' ||
      variables.androidRipple === false ||
      Platform['Version'] < 21
    ) {
      return (
        <TouchableOpacity
          {...this.prepareRootProps()}
          ref={(c) => (this._root = c)}
          activeOpacity={this.props.activeOpacity > 0 ? this.props.activeOpacity : 0.5}>
          {children}
        </TouchableOpacity>
      )
    } else {
      if (this.props.rounded) {
        let buttonStyle = { ...this.prepareRootProps().style }
        let buttonFlex = this.props.full || this.props.block ? 1 : buttonStyle.flex
        return (
          <View
            style={[
              { maxHeight: buttonStyle.height },
              buttonStyle,
              { paddingTop: undefined, paddingBottom: undefined }
            ]}>
            <TouchableNativeFeedback
              ref={(c) => (this._root = c)}
              background={
                this.props.androidRippleColor
                  ? TouchableNativeFeedback.Ripple(this.props.androidRippleColor, true)
                  : TouchableNativeFeedback.Ripple(variables.androidRippleColor, true)
              }
              {...this.prepareRootProps()}>
              <View
                style={[
                  styles.childContainer,
                  {
                    paddingTop: buttonStyle.paddingTop,
                    paddingBottom: buttonStyle.paddingBottom,
                    height: buttonStyle.height,
                    flexGrow: buttonFlex
                  }
                ]}>
                {children}
              </View>
            </TouchableNativeFeedback>
          </View>
        )
      } else {
        return (
          <TouchableNativeFeedback
            ref={(c) => (this._root = c)}
            onPress={this.props.onPress}
            background={
              this.props.transparent
                ? TouchableNativeFeedback.Ripple('transparent')
                : TouchableNativeFeedback.Ripple(variables.androidRippleColor, false)
            }
            {...this.prepareRootProps()}>
            <View {...this.prepareRootProps()}>{children}</View>
          </TouchableNativeFeedback>
        )
      }
    }
  }
}

// Button.propTypes = {
//   ...TouchableOpacity.propTypes,
//   style: PropTypes.oneOfType([
//     PropTypes.object,
//     PropTypes.number,
//     PropTypes.array
//   ]),
//   block: PropTypes.bool,
//   primary: PropTypes.bool,
//   transparent: PropTypes.bool,
//   success: PropTypes.bool,
//   danger: PropTypes.bool,
//   warning: PropTypes.bool,
//   info: PropTypes.bool,
//   bordered: PropTypes.bool,
//   disabled: PropTypes.bool,
//   rounded: PropTypes.bool,
//   large: PropTypes.bool,
//   small: PropTypes.bool,
//   active: PropTypes.bool
// };

const styles = StyleSheet.create({
  childContainer: {
    flexShrink: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  }
})

const StyledButton = connectStyle('NativeBase.Button', {}, mapPropsToStyleNames)(Button)
export { StyledButton as Button }
