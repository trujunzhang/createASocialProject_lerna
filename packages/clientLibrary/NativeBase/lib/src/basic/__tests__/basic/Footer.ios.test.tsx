import 'react-primitives'
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { Container } from '../../Container'
import { Header } from '../../Header'
import { Button } from '../../Button'
import { Icon } from '../../Icon'
import { Content } from '../../Content'
import { Text } from '../../Text'
import { Footer } from '../../Footer'
import { FooterTab } from '../../FooterTab'
import { Badge } from '../../Badge'

// Note: test renderer must be required after react-native.

// jest.mock("Platform", () => {
//   const Platform = require.requireActual("Platform");
//   Platform.OS = "ios";
//  return Platform;
// });

it('renders basic Footer', () => {
  const tree = renderer
    .create(
      <Container>
        <Header />
        <Content />
        <Footer>
          <FooterTab>
            <Button active={true}>
              <Text>Apps</Text>
            </Button>
            <Button active={false}>
              <Text>Camera</Text>
            </Button>
            <Button active={false}>
              <Text>Navigate</Text>
            </Button>
            <Button active={false}>
              <Text>Contact</Text>
            </Button>
          </FooterTab>
        </Footer>
      </Container>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Footer with Icons', () => {
  const tree = renderer
    .create(
      <Container>
        <Header />
        <Content />
        <Footer>
          <FooterTab>
            <Button active={true}>
              <Icon active={true} name="apps" />
            </Button>
            <Button active={false}>
              <Icon active={false} name="camera" />
            </Button>
            <Button active={false}>
              <Icon active={false} name="compass" />
            </Button>
            <Button active={false}>
              <Icon active={false} name="contact" />
            </Button>
          </FooterTab>
        </Footer>
      </Container>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Footer with Icon and Text', () => {
  const tree = renderer
    .create(
      <Container>
        <Header />
        <Content />
        <Footer>
          <FooterTab>
            <Button active={true}>
              <Icon active={true} name="apps" />
              <Text>Apps</Text>
            </Button>
            <Button active={false}>
              <Icon active={false} name="camera" />
              <Text>Camera</Text>
            </Button>
            <Button active={false}>
              <Icon active={false} name="compass" />
              <Text>Compass</Text>
            </Button>
            <Button active={false}>
              <Icon active={false} name="contact" />
              <Text>Contact</Text>
            </Button>
          </FooterTab>
        </Footer>
      </Container>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Footer with Badge', () => {
  const tree = renderer
    .create(
      <Container>
        <Header />
        <Content />
        <Footer>
          <FooterTab>
            <Button active={true} vertical badge>
              <Badge>
                <Text>2</Text>
              </Badge>
              <Icon active={true} name="apps" />
              <Text>Apps</Text>
            </Button>
            <Button active={false}>
              <Icon active={false} name="camera" />
              <Text>Camera</Text>
            </Button>
            <Button active={false} vertical badge>
              <Badge style={{ backgroundColor: 'green' }}>
                <Text>51</Text>
              </Badge>
              <Icon active={false} name="compass" />
              <Text>Compass</Text>
            </Button>
            <Button active={false}>
              <Icon active={false} name="contact" />
              <Text>Contact</Text>
            </Button>
          </FooterTab>
        </Footer>
      </Container>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
