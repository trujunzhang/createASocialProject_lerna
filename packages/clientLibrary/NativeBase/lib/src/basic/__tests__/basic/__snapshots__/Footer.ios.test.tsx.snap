// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Footer with Badge 1`] = `
<view
  name="container"
  style={Object {}}
>
  <view
    name="header"
  >
    <view
      name="statusBar"
    />
    <view
      name="header-other"
      style={Object {}}
    />
  </view>
  <view
    name="keyboardAwareScrollView"
    style={Object {}}
  />
  <view
    name="View"
    style={Object {}}
  >
    <view
      name="View"
      style={Object {}}
    >
      <view
        name="TouchableOpacity: "
      />
      <view
        name="TouchableOpacity: "
      />
      <view
        name="TouchableOpacity: "
      />
      <view
        name="TouchableOpacity: "
      />
    </view>
  </view>
</view>
`;

exports[`renders Footer with Icon and Text 1`] = `
<view
  name="container"
  style={Object {}}
>
  <view
    name="header"
  >
    <view
      name="statusBar"
    />
    <view
      name="header-other"
      style={Object {}}
    />
  </view>
  <view
    name="keyboardAwareScrollView"
    style={Object {}}
  />
  <view
    name="View"
    style={Object {}}
  >
    <view
      name="View"
      style={Object {}}
    >
      <view
        name="TouchableOpacity: "
      />
      <view
        name="TouchableOpacity: "
      />
      <view
        name="TouchableOpacity: "
      />
      <view
        name="TouchableOpacity: "
      />
    </view>
  </view>
</view>
`;

exports[`renders Footer with Icons 1`] = `
<view
  name="container"
  style={Object {}}
>
  <view
    name="header"
  >
    <view
      name="statusBar"
    />
    <view
      name="header-other"
      style={Object {}}
    />
  </view>
  <view
    name="keyboardAwareScrollView"
    style={Object {}}
  />
  <view
    name="View"
    style={Object {}}
  >
    <view
      name="View"
      style={Object {}}
    >
      <view
        name="View"
        style={
          Object {
            "borderBottomLeftRadius": 2,
            "borderBottomRightRadius": 2,
            "borderTopLeftRadius": 2,
            "borderTopRightRadius": 2,
          }
        }
      >
        <view
          name="View"
          style={Object {}}
        >
          <text
            style={Object {}}
          >
            ios-apps
          </text>
        </view>
      </view>
      <view
        name="View"
        style={
          Object {
            "borderBottomLeftRadius": 2,
            "borderBottomRightRadius": 2,
            "borderTopLeftRadius": 2,
            "borderTopRightRadius": 2,
          }
        }
      >
        <view
          name="View"
          style={Object {}}
        >
          <text
            style={Object {}}
          >
            ios-camera
          </text>
        </view>
      </view>
      <view
        name="View"
        style={
          Object {
            "borderBottomLeftRadius": 2,
            "borderBottomRightRadius": 2,
            "borderTopLeftRadius": 2,
            "borderTopRightRadius": 2,
          }
        }
      >
        <view
          name="View"
          style={Object {}}
        >
          <text
            style={Object {}}
          >
            ios-compass
          </text>
        </view>
      </view>
      <view
        name="View"
        style={
          Object {
            "borderBottomLeftRadius": 2,
            "borderBottomRightRadius": 2,
            "borderTopLeftRadius": 2,
            "borderTopRightRadius": 2,
          }
        }
      >
        <view
          name="View"
          style={Object {}}
        >
          <text
            style={Object {}}
          >
            ios-contact
          </text>
        </view>
      </view>
    </view>
  </view>
</view>
`;

exports[`renders basic Footer 1`] = `
<view
  name="container"
  style={Object {}}
>
  <view
    name="header"
  >
    <view
      name="statusBar"
    />
    <view
      name="header-other"
      style={Object {}}
    />
  </view>
  <view
    name="keyboardAwareScrollView"
    style={Object {}}
  />
  <view
    name="View"
    style={Object {}}
  >
    <view
      name="View"
      style={Object {}}
    >
      <view
        name="View"
        style={
          Object {
            "borderBottomLeftRadius": 2,
            "borderBottomRightRadius": 2,
            "borderTopLeftRadius": 2,
            "borderTopRightRadius": 2,
          }
        }
      >
        <view
          name="View"
          style={Object {}}
        >
          <text
            style={Object {}}
          >
            Apps
          </text>
        </view>
      </view>
      <view
        name="View"
        style={
          Object {
            "borderBottomLeftRadius": 2,
            "borderBottomRightRadius": 2,
            "borderTopLeftRadius": 2,
            "borderTopRightRadius": 2,
          }
        }
      >
        <view
          name="View"
          style={Object {}}
        >
          <text
            style={Object {}}
          >
            Camera
          </text>
        </view>
      </view>
      <view
        name="View"
        style={
          Object {
            "borderBottomLeftRadius": 2,
            "borderBottomRightRadius": 2,
            "borderTopLeftRadius": 2,
            "borderTopRightRadius": 2,
          }
        }
      >
        <view
          name="View"
          style={Object {}}
        >
          <text
            style={Object {}}
          >
            Navigate
          </text>
        </view>
      </view>
      <view
        name="View"
        style={
          Object {
            "borderBottomLeftRadius": 2,
            "borderBottomRightRadius": 2,
            "borderTopLeftRadius": 2,
            "borderTopRightRadius": 2,
          }
        }
      >
        <view
          name="View"
          style={Object {}}
        >
          <text
            style={Object {}}
          >
            Contact
          </text>
        </view>
      </view>
    </view>
  </view>
</view>
`;
